### PR TITLE
Refresh AKS Onboarding ARM Template Scenario

### PR DESCRIPTION
**Pull request by zaidmohd**
Updated Kubernetes version to 1.23.5
Update AKS ARM template API version
Updated script to include providers registrations
Updated Readme

### CODE REVIEW

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md [hunk 1]
```diff
@@ -10,25 +10,25 @@ description: >
 
 The following Jumpstart scenario will guide you on how to use the provided [Azure ARM Template](https://docs.microsoft.com/azure/azure-resource-manager/templates/overview) to deploy an [Azure Kubernetes Service (AKS)](https://docs.microsoft.com/azure/aks/intro-kubernetes) cluster and connected it as an Azure Arc cluster resource.
 
-  > **NOTE: Since AKS is a 1st-party Azure solution and natively supports capabilities such as [Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/insights/container-insights-overview) integration as well as GitOps configurations (currently in preview), it is not expected for an AKS cluster to be projected as an Azure Arc-enabled Kubernetes cluster. The following scenario should ONLY be used for demo and testing purposes.**
+  > **NOTE: Since AKS is a 1st-party Azure solution and natively supports capabilities such as [Azure Monitor](https://docs.microsoft.com/azure/azure-monitor/insights/container-insights-overview) integration as well as GitOps configurations, it is not expected for an AKS cluster to be projected as an Azure Arc-enabled Kubernetes cluster. The following scenario should ONLY be used for demo and testing purposes.**
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md:13]: It used to be the case but I think we should also incl. data services, app services, and ML. Something similar to the message you are getting when onboarding AKS.

File: azure_arc_k8s_jumpstart/aks/arm_template/scripts/az_connect_aks.sh [hunk 2]
```diff
@@ -9,11 +9,25 @@ export tenantId='<Your Azure tenant ID>'
 export resourceGroup='<Azure resource group name>'
 export arcClusterName='<The name of your k8s cluster as it will be shown in Azure Arc>'
 
-# Getting AKS credentials
-echo "Log in to Azure with Service Principle & Getting AKS credentials (kubeconfig)"
+# Login to Azure
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/aks/arm_template/scripts/az_connect_aks.sh:12]:
```suggestion
# Log in to Azure
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md [hunk 3]
```diff
@@ -111,33 +95,47 @@ The following Jumpstart scenario will guide you on how to use the provided [Azur
...
 
-* In order to keep your local environment clean and untouched, we will use [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview) (located in the top-right corner in the Azure portal) to run the *az_connect_aks* shell script against the AKS cluster. **Make sure Cloud Shell is configured to use Bash.**
+  For example:
 
-  ![Screenshot of Azure Cloud Shell button in Visual Studio Code](./05.png)
+  ![Screenshot of az_connect_aks shell script](./05.png)
 
-* After editing the environment variables in the [*az_connect_aks*](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/aks/arm_template/scripts/az_connect_aks.sh) shell script to match your parameters, save the file and then upload it to the Cloud Shell environment and run it using the ```. ./az_connect_aks.sh``` command.
+- In order to keep your local environment clean and untouched, we will use [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview) (located in the top-right corner in the Azure portal) to run the *az_connect_aks* shell script against the AKS cluster. **Make sure Cloud Shell is configured to use Bash.**
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md:126]:
```suggestion
- In order to keep your local environment clean and untouched, we will use [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview) (located in the top-right corner of the Azure portal) to run the *az_connect_aks* shell script against the AKS cluster. **Make sure Cloud Shell is configured to use Bash.**
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md [hunk 4]
```diff
@@ -111,33 +95,47 @@ The following Jumpstart scenario will guide you on how to use the provided [Azur
...
 
   ![Screenshot showing upload of file to Cloud Shell](./07.png)
 
-  ![Screenshot showing Cloud Shell with uploaded file](./08.png)
+  ![Screenshot showing upload of file to Cloud Shell](./08.png)
 
-* Once the script run has finished, the AKS cluster will be projected as a new Azure Arc cluster resource.
+- Once the script run has finished, the AKS cluster will be projected as a new Azure Arc cluster resource.
 
   ![Screenshot showing Azure Portal with Azure Arc-enabled Kubernetes resource](./09.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md:140]:
```suggestion
  ![Screenshot showing Azure portal with Azure Arc-enabled Kubernetes resource](./09.png)
```

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md [hunk 5]
```diff
@@ -111,33 +95,47 @@ The following Jumpstart scenario will guide you on how to use the provided [Azur
   --parameters azuredeploy.parameters.json
   ```
 
-* Once the ARM template deployment is completed, a new AKS cluster in a new Azure resource group is created.
+- Once the ARM template deployment is completed, a new AKS cluster in a new Azure resource group is created.
 
   ![Screenshot of Azure Portal showing AKS resource](./02.png)
 
   ![Screenshot of Azure Portal showing AKS resource](./03.png)
```
Comment (likamrat) [inline docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md:102]:
```suggestion
  ![Screenshot of Azure portal showing AKS resource](./02.png)

  ![Screenshot of Azure portal showing AKS resource](./03.png)
```

File: azure_arc_k8s_jumpstart/aks/arm_template/azuredeploy.parameters.json [hunk 6]
```diff
@@ -21,7 +21,7 @@
             "value": "<Your Azure service principal password>"
         },
         "kubernetesVersion": {
-            "value": "<Kubernetes Version, e.g. 1.18.2>"
+            "value": "<Kubernetes Version, e.g. 1.23.5>"
         }
     }
 }
```
Comment (likamrat) [inline azure_arc_k8s_jumpstart/aks/arm_template/azuredeploy.parameters.json:27]: Not shown in the comment, but I think we can drop the AKS prefix from the parameters file and set it as default in the deployment template. [See example}(https://github.com/microsoft/azure_arc/blob/main/azure_arc_data_jumpstart/aks/ARM/azuredeploy.json#L69)

**Review by likamrat — CHANGES_REQUESTED** [review]
- DNS prefix suggestion
- AKS onboarding note suggestion
- Minor text suggestions

File: azure_arc_k8s_jumpstart/aks/arm_template/azuredeploy.parameters.json (hunk 6 above)
Comment (zaidmohd) [reply azure_arc_k8s_jumpstart/aks/arm_template/azuredeploy.parameters.json:27]: Added default value for dns prefix in main template.
Removed dnsprefix from parameters file and screenshot.

File: docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md (hunk 1 above)
Comment (zaidmohd) [reply docs/azure_arc_jumpstart/azure_arc_k8s/aks/aks_arm_template/_index.md:13]: Updated

**Review by likamrat — APPROVED** [review]
LGTM. Great work!

**Pull request merged**